wicker/core/storage.py

```diff
@@ -44,6 +45,28 @@ def fetch_file(self, input_path: str, local_prefix: str, timeout_seconds: int) -
         """
         pass
 
```

Comment:
Below is the main refactoring in this PR, i.e. add `put_file` and `put_object` to the `AbstractDataStorage` interface.